test/ntp_client_test.rb

```diff
@@ -0,0 +1,487 @@
+      expect(Yast::Mode).not_to receive(:normal)
+
+      expect(subject.Read).to eql(true)
+    end
+
+    it "returns false if abort is pressed" do
+      allow(subject).to receive(:Abort).and_return(true)
+
+      expect(subject.Read).to eql(false)
+    end
+
+    it "doesn't show progress if it is not in normal Mode" do
+      expect(Yast::Progress).not_to receive(:New)
+
+      subject.Read
+    end
+
+    it "reads network interfaces config" do
+      expect(Yast::NetworkInterfaces).to receive(:Read)
+
+      subject.Read
+    end
+
+    it "reads ntp policy setting, using auto as default if not exists" do
+      expect(Yast::SCR).to receive(:Read)
+        .with(Yast::Path.new(".sysconfig.network.config.NETCONFIG_NTP_POLICY"))
+        .and_return(nil)
+      expect(subject.ntp_policy).to eql("auto")
+
+      subject.Read
+    end
+
+    it "loads known ntp servers and known country names" do
+      expect(subject).to receive(:GetNtpServers)
+      expect(subject).to receive(:GetCountryNames)
+
+      subject.Read
+    end
+
+    it "returns false if the ntp package is not and could not be installed" do
+      expect(Yast::PackageSystem).to receive(:CheckAndInstallPackagesInteractive)
+        .with(["ntp"]).and_return(false)
+      expect(Yast::Service).not_to receive(:Enabled)
+
+      expect(subject.Read).to eql(false)
+    end
+
+    it "checks if ntpd service is enable" do
+      expect(Yast::Service).to receive(:Enabled).with("ntpd")
+
+      subject.Read
+    end
+
+    it "checks active directory's ntp dumped data file" do
+      expect(Yast::FileUtils).to receive(:Exists).with("#{Yast::Directory.vardir}/ad_ntp_data.ycp")
+
+      subject.Read
+    end
+
+    it "reads ntp config from /etc/ntp.conf" do
+      expect(subject).to receive(:ProcessNtpConf)
+
+      subject.Read
+    end
+
+    it "reads synchronization config" do
+      expect(subject).to receive(:ReadSynchronization)
+
+      subject.Read
+    end
+
+    it "reads ntpd chroot configuration" do
+      expect(Yast::SCR).to receive(:Read)
+        .with(Yast::Path.new(".sysconfig.ntp.NTPD_RUN_CHROOTED"))
+
+      subject.Read
+    end
+
+    it "returns true if all reads were performed" do
+      expect(subject.Read).to eql(true)
+    end
+  end
+
+  describe "#Write" do
+    let(:data_dir) { File.join(File.dirname(__FILE__), "data") }
+
+    around do |example|
+      ::FileUtils.cp(File.join(data_dir, "scr_root/etc/ntp.conf.original"),
+        File.join(data_dir, "scr_root/etc/ntp.conf"))
+      change_scr_root(File.join(data_dir, "scr_root"), &example)
+    end
+
+    before do
+      allow(Yast::Mode).to receive(:normal).and_return(false)
+      allow(subject).to receive(:Abort).and_return(false)
+      allow(Yast::SCR).to receive(:Read)
+        .with(Yast::Path.new(".etc.ntp_conf.all")).and_return(nil)
+      allow(Yast::FileChanges).to receive(:StoreFileCheckSum).with("/etc/ntp.conf")
+      allow(Yast::SCR).to receive(:Write)
```

Comment:
Just asking: do you need to mock this call? You're using a `scr_root/etc/sysconfig/network` directory, right?